ci: the release-v3.10 branch needs Kubernetes 1.28 jobs

nixpanic opened the pull request:
With Ceph-CSI v3.10 we do not test Kubernetes 1.25 anymore, and we
expect Kubernetes 1.28 to be used.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
ci: the release-v3.10 branch needs Kubernetes 1.28 jobs

With Ceph-CSI v3.10 we do not test Kubernetes 1.25 anymore, and we
expect Kubernetes 1.28 to be used.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

Rakshith-R commented on .mergify.yml:
```diff
@@ -34,6 +34,25 @@ queue_rules:
+              - "status-success=ci/centos/upgrade-tests-cephfs"
+              - "status-success=ci/centos/upgrade-tests-rbd"
+          - and:
+              - base~=release-v3.[89]
```
let's ignore release v3.8 since it will be deprecated when 3.10 comes out ?

nixpanic replied:
sure, removed it.

Rakshith-R reviewed (commented):
Thanks for the changes,
left a comment regarding leaving v3.8 branch out

nixpanic committed:
ci: use regex comparing `~=` for `release-v*` branch checking

It seems that `base=release-v*` does not match the `release-v3.10`
branch for some reason. Maybe a Mergify update requires stricter
checking of strings, and only allows the `~=` operator for regular
expressions now?

Regular expressions like `base~=^(devel)|(release-.+)$` seem to match
for `release-3.10`, so let's use that notation everywhere.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

Rakshith-R reviewed (approved):
Thanks

nixpanic commented:
@mergifyio queue
